File: TransCoder_tokenized_test_set_functions/java/FIND_THE_NEAREST_SMALLER_NUMBERS_ON_LEFT_SIDE_IN_AN_ARRAY.java
static void printPrevSmaller ( int [ ] arr, int n ) {
  System . out . print ( "_, " );
  for ( int i = 1;
  i < n;
  i ++ ) {
    int j;
    for ( j = i - 1;
    j >= 0;
    j -- ) {
      if ( arr [ j ] < arr [ i ] ) {
        System . out . print ( arr [ j ] + ", " );
        break;
      }
    }
    if ( j == - 1 ) System . out . print ( "_, " );
  }
}

File: TransCoder_tokenized_test_set_functions/python/FIND_THE_NEAREST_SMALLER_NUMBERS_ON_LEFT_SIDE_IN_AN_ARRAY.py
def printPrevSmaller ( arr, n ) :
    print ( "_, ", end = "" )
    for i in range ( 1, n ) :
        for j in range ( i - 1, - 2, - 1 ) :
            if ( arr [ j ] < arr [ i ] ) :
                print ( arr [ j ], ", ", end = "" )
                break
        if ( j == - 1 ) :
            print ( "_, ", end = "" )
    
